pkg/processor/runtime/python/py/_nuclio_wrapper_as_server.py

```diff
@@ -0,0 +1,308 @@
+
+            if self._is_termination_needed:
+                result = self._call_termination_handler()
+                if asyncio.iscoroutine(result):
+                    await result
+            self._cleanup_connection(sock, cancel_task=False)
+
+    async def _handle_event(self, event, sock):
+        # take call time
+        start_time = time.time()
+
+        # call the entrypoint
+        entrypoint_output = await self._entrypoint(self._context, event)
+
+        # measure duration, set to minimum float in case execution was too fast
+        duration = time.time() - start_time or sys.float_info.min
+
+        await self._write_packet_to_processor(sock, 'm' + json.dumps({'duration': duration}))
+
+        # try to json encode the response
+        encoded_response = self._encode_entrypoint_output(entrypoint_output)
+
+        # write response to the socket
+        await self._write_packet_to_processor(sock, 'r' + encoded_response)
+
+    def _cleanup_connection(self, sock, cancel_task=True):
+        """Cleanup resources for a disconnected client."""
+        fileno = sock.fileno()
+        if fileno in self.connections:
+            del self.connections[fileno]
+            self._logger.info(f"Connection {fileno} removed from tracking")
+        sock.close()
+
+        # Cancel and remove the associated task
+        # do not cancel if calling from the task itself
+        if cancel_task and fileno in self.tasks:
+            task = self.tasks.pop(fileno)
+            if not task.done():
+                task.cancel()
+            self._logger.info(f"Task for connection {fileno} cancelled and removed")
+
+    async def _shutdown(self, error_code=0):
+        """Shutdown the server and cleanup resources."""
+        self._logger.info("Shutting down...")
+        try:
+            await self._stop_processing()
+        finally:
+            super()._shutdown()
+
+    async def _stop_processing(self):
+        self.shutdown_event.set()
+
+        # Cancel all active tasks
+        for fileno, task in list(self.tasks.items()):
+            self._logger.info(f"Cancelling task for connection {fileno}")
+            task.cancel()
+        # Close all active sockets
+        for fileno, sock in list(self.connections.items()):
+            self._logger.info(f"Closing connection {fileno}")
+            sock.close()
+        self.selector.close()
+
+
+def parse_args():
+    parser = get_parser_with_common_args()
+    return parser.parse_args()
+
+
+def run_wrapper():
+    # parse arguments
+    args = parse_args()
+
+    # create a logger instance. note: there are no outputters until socket is created
+    root_logger = create_logger(args.log_level)
+
+    # add a logger output that is in a JSON format. we'll remove it once we have a socket output. this
+    # way all output goes to stdout until a socket is available and then switches exclusively to socket
+    root_logger.set_handler('default', sys.stdout, nuclio_sdk.logger.JSONFormatter())
```

Comment:
On the same logger note, I noticed that in this wrapper we don't replace the logger handler with the socket file after we connect to it.
It makes sense as you now have multiple sockets, but does it have any effect on how the logs are processed?
If not - do we really need to do this in the original wrapper?

Reply:
https://iguazio.atlassian.net/browse/NUC-361